src/mms_images_extractor.py

```diff
@@ -10,106 +10,132 @@
 
 
 def get_datetime_from_epoch_milliseconds(epoch_milliseconds: str) -> str:
-    """
-    Converts the epoch time in milliseconds to a human-readable date and time string
-    """
     return datetime.datetime.fromtimestamp(int(epoch_milliseconds) / 1000).strftime('%Y%m%d-%H%M%S')
 
 
-def reconstruct_sms_images(sms_xml_dir: dir, output_images_dir: str) -> None:
+def safe_filename(base_dir: str, filename: str, max_len=200) -> str:
+    full_path = os.path.join(base_dir, filename)
+    if len(full_path) <= 255:
```

Comment:
Minor note, I would create a constant var for this value and use that instead rather than hardcoding it in 2 places.